uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/DrupalUserService.php

```diff
@@ -59,19 +46,22 @@ public function processUserAccount($user, $roles, $params = []) {
      * UFMatch is created by CiviCRM function civicrm_user_insert listening
      * for user_module_invoke('insert') from Drupal
      */
-    $user = user_save($user, $params);
+    $user = user_save(drupal_anonymous_user(), $params);
 
     return $user;
   }
 
   /**
-   * Finds a user record by email
+   * Checks if user record exist by email
    *
    * @param string $email
-   * @return array|boolean
+   *
+   * @return boolean
    */
-  public function findUserByEmail($email) {
-    return user_load_by_mail($email);
+  public function userExistsForEmail($email) {
```

Comment:
Please see sample tests as a guide to write test for this method in this classes: https://github.com/compucorp/civihr/blob/9313c6246af92409a664840f6574031edd9a169d/uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/CMSData/UserAccount/DrupalTest.php#L2-L1 .

Drupal is not bootstrapped during Civi tests, so what we do is to have a file that mocks drupal functions and the return results. For the test for this function, you can modify this function here: https://github.com/compucorp/civihr/blob/16324ac5eb8033d87beac6bf1dabb58751528501/uk.co.compucorp.civicrm.hrcore/tests/phpunit/drupal_function_mocks.php#L13-L12, to return TRUE for a particular email and false for any other email, that, way you can test that the function returns true or false.